File: src/Moryx.FactoryMonitor.Endpoints/Model/CellDataModel.cs
// Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Moryx.FactoryMonitor.Endpoints.Model
{
    /// <summary>
    /// Model containing all the data about a resource (Activity, process,state etc..)
    /// </summary>
    [DataContract]
    [Obsolete("Use CellStateChanged model instead")]
    public class CellDataModel
    {      
        [DataMember]
        public virtual CellState State { get; set; }

        [DataMember]
        public virtual ProcessModel ProcessActivity { get; set; }

        [DataMember]
        public virtual List<TransportPathModel> ForkliftDispatchRoutes { get; set; }
    }
}

File: src/Moryx.FactoryMonitor.Web/src/app/api/models/Moryx/FactoryMonitor/Endpoints/Model/cell-data-model.ts
/* tslint:disable */
/* eslint-disable */
import { CellState as MoryxFactoryMonitorEndpointsModelCellState } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/cell-state';
import { ProcessModel as MoryxFactoryMonitorEndpointsModelProcessModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/process-model';
import { TransportPathModel as MoryxFactoryMonitorEndpointsModelTransportPathModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/transport-path-model';
export interface CellDataModel {
  forkliftDispatchRoutes?: null | Array<MoryxFactoryMonitorEndpointsModelTransportPathModel>;
  processActivity?: MoryxFactoryMonitorEndpointsModelProcessModel;
  state?: MoryxFactoryMonitorEndpointsModelCellState;
}
